tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/label/LabelChangeRepository.scala

```diff
@@ -0,0 +1,139 @@
+package com.linagora.tmail.james.jmap.label
+
+import com.google.common.collect.{HashBasedTable, Table, Tables}
+import com.linagora.tmail.james.jmap.model.LabelId
+import org.apache.james.jmap.api.change.{JmapChange, Limit, State}
+import org.apache.james.jmap.api.exception.ChangeNotFoundException
+import org.apache.james.jmap.api.model.AccountId
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.time.ZonedDateTime
+import javax.inject.Named
+
+trait LabelChangeRepository {
+
+
+  def save(labelChange: LabelChange): Publisher[Void]
+
+  def getSinceState(accountId: AccountId, state: State, maxIdsToReturn: Option[Limit] = None): Publisher[LabelChanges]
+
+  def getLatestState(accountId: AccountId): Publisher[State]
+
+  def getLatestStateWithDelegation(accountId: AccountId): Publisher[State]
+
+  def getSinceStateWithDelegation(accountId: AccountId, state: State, maxIdsToReturn: Option[Limit] = None): Publisher[LabelChanges]
+}
+
+case class LabelChange(accountId: AccountId,
+                       created: Set[LabelId],
+                       updated: Set[LabelId],
+                       destroyed: Set[LabelId],
+                       newState: State,
+                       hasMoreChanges: Boolean,
+                       isDelegated: Boolean,
+                       date: ZonedDateTime) extends JmapChange {
+  override def getAccountId: AccountId = accountId
+}
+
+case class LabelChanges(newState: State,
+                        hasMoreChanges: Boolean,
+                        created: Set[LabelId],
+                        updated: Set[LabelId],
+                        destroyed: Set[LabelId]) {
+  def getAllChanges: Set[LabelId] = created ++ updated ++ destroyed
+
+  def append(labelChange: LabelChange): LabelChanges =
+    this.copy(created = created ++ labelChange.created,
+      updated = updated ++ labelChange.updated,
+      destroyed = destroyed ++ labelChange.destroyed)
+}
+
+
+case class LabelChangesBuilder(var created: Set[LabelId] = Set(),
```

Comment:
Builders are non idiomatic in Scala.

Make it a method in LabelChanges case class or object